File: java/simpleTest.java
import coppelia.b0RemoteApi;

import org.msgpack.core.MessageUnpacker;
import org.msgpack.value.*;

import java.io.IOException;
import java.io.UncheckedIOException;
import java.util.Map;

public class simpleTest
{
    public static boolean doNextStep=true;
    public static int visionSensorHandle;
    public static int passiveVisionSensorHandle;
    public static b0RemoteApi client;
    
    public static void simulationStepStarted(final MessageUnpacker msg)
    {
        try
        {
            Map<Value,Value> map=client.readValue(msg,1).asMapValue().map();
            float simTime=map.get(ValueFactory.newString("simulationTime")).asNumberValue().toFloat();
            System.out.println("Simulation step started. Simulation time: "+simTime);
        }
        catch(IOException e) { throw new UncheckedIOException(e); }
    }
        
    public static void simulationStepDone(final MessageUnpacker msg)
    {
        try
        {
            Map<Value,Value> map=client.readValue(msg,1).asMapValue().map();
            float simTime=map.get(ValueFactory.newString("simulationTime")).asNumberValue().toFloat();
            System.out.println("Simulation step done. Simulation time: "+simTime);
            doNextStep=true;
        }
        catch(IOException e) { throw new UncheckedIOException(e); }
    }
        
    public static void imageCallback(final MessageUnpacker msg)
    {
        try
        {
            byte[] img=client.readByteArray(msg,2);
            client.simxSetVisionSensorImage(passiveVisionSensorHandle,false,img,client.simxDefaultPublisher());
            System.out.println("Received image.");
        }
        catch(IOException e) { throw new UncheckedIOException(e); }
    }
        
    public static void main(String[] args) throws IOException
    {
        
        client = new b0RemoteApi();
        client.simxAddStatusbarMessage("Hello world!",client.simxDefaultPublisher());
        MessageUnpacker msg=client.simxGetObjectHandle("VisionSensor",client.simxServiceCall());
        visionSensorHandle=client.readInt(msg,1);
        msg=client.simxGetObjectHandle("PassiveVisionSensor",client.simxServiceCall());
        passiveVisionSensorHandle=client.readInt(msg,1);
        client.simxSynchronous(true);
        
        client.simxGetVisionSensorImage(visionSensorHandle,false,client.simxDefaultSubscriber(simpleTest::imageCallback));
        client.simxGetSimulationStepStarted(client.simxDefaultSubscriber(simpleTest::simulationStepStarted));
        client.simxGetSimulationStepDone(client.simxDefaultSubscriber(simpleTest::simulationStepDone));
        client.simxStartSimulation(client.simxDefaultPublisher());
        
        long startTime = System.currentTimeMillis();
        while (System.currentTimeMillis()<startTime+5000)
        {
            if (doNextStep)
            {
                doNextStep=false;
                client.simxSynchronousTrigger();
            }
            client.simxSpinOnce();
        }
        client.simxStopSimulation(client.simxDefaultPublisher());
        
        client.close();
        System.out.println("Program ended");
    }
}

File: python/simpleTest.py
import b0RemoteApi
import time

with b0RemoteApi.RemoteApiClient('b0RemoteApi_pythonClient','b0RemoteApi') as client:    
    doNextStep=True

    def simulationStepStarted(msg):
        simTime=msg[1]['simulationTime'];
        print('Simulation step started. Simulation time: ',simTime)
        
    def simulationStepDone(msg):
        simTime=msg[1]['simulationTime'];
        print('Simulation step done. Simulation time: ',simTime);
        global doNextStep
        doNextStep=True
        
    def imageCallback(msg):
        print('Received image.',msg[1])
        client.simxSetVisionSensorImage(passiveVisionSensorHandle[1],False,msg[2],client.simxDefaultPublisher())
    
    client.simxAddStatusbarMessage('Hello world!',client.simxDefaultPublisher())
    visionSensorHandle=client.simxGetObjectHandle('VisionSensor',client.simxServiceCall())
    passiveVisionSensorHandle=client.simxGetObjectHandle('PassiveVisionSensor',client.simxServiceCall())
    client.simxSynchronous(True)
    
#    dedicatedSub=client.simxCreateSubscriber(imageCallback)
#    client.simxGetVisionSensorImage(visionSensorHandle[1],False,dedicatedSub)
    client.simxGetVisionSensorImage(visionSensorHandle[1],False,client.simxDefaultSubscriber(imageCallback))

    client.simxGetSimulationStepStarted(client.simxDefaultSubscriber(simulationStepStarted));
    client.simxGetSimulationStepDone(client.simxDefaultSubscriber(simulationStepDone));
    client.simxStartSimulation(client.simxDefaultPublisher())
    
    startTime=time.time()
    while time.time()<startTime+5: 
        if doNextStep:
            doNextStep=False
            client.simxSynchronousTrigger()
        client.simxSpinOnce()
    client.simxStopSimulation(client.simxDefaultPublisher())
